src/test/java/de/tum/cit/aet/artemis/buildagent/service/CustomFeedbackParserTest.java

```
@@ -143,4 +143,12 @@ void shouldNotParseCustomFeedbackWithNullName() {
 
         assertThat(successfulTests).isEmpty();
     }
+
+    @Test
+    void shouldNotParseInvalidJson() {
+        final var fileContent = "{ invalid json }";
+        CustomFeedbackParser.processTestResultFile(fileName, fileContent, failedTests, successfulTests);
+        assertThat(failedTests).isEmpty();
```

Is this really desired behavior? Wouldn't it be better to fail/throw an exception in case of invalid JSON.

---

Would an exception cause the system to crash? Or would it be caught by the system?

---

No, this won't crash the server application.

---

Ok, then should I delete this method?

---

Hi, I deleted this method; I'd appreciate it if you could take another look and approve the PR.

---

Yeah, if the system will know if the JSON is invalid, then there is no need to test it. So, I went ahead and removed this test.

---

I don't really get what you mean with "the system"